src/guide/essentials/template-refs.md

```diff
@@ -214,11 +215,11 @@ export default {
 
 </div>
 
-<span class="composition-api">If the child component is using Options API or not using `<script setup>`, the</span><span class="options-api">The</span> referenced instance will be identical to the child component's `this`, which means the parent component will have full access to every property and method of the child component. This makes it easy to create tightly coupled implementation details between the parent and the child, so component refs should be only used when absolutely needed - in most cases, you should try to implement parent / child interactions using the standard props and emit interfaces first.
+<span class="composition-api">Agar bola komponent Options API ishlatayotgan bo'lsa yoki `<script setup>` ishlatmasa, bog'langan</span><span class="options-api">Bog'langan</span> namuna bola komponentning `this` kalit so'zi bilan bir xil bo'ladi, shuningdek ota komponent bola komponentning har bir xususiyati(property) va metodlariga to'liq ulana oladi - foydalana oladi. Bu esa ota va bola komponent o'rtasidagi birgalikda amalga oshiriladigan chambarchas ma'lumotlarni yaratishni osonlashtiradi, shuning uchun ko'p holatlarda komponent reflar kerakli bo'lganda ishlatilishi kerak. Birinchi o'rinda standard xususiyatlar(props) va emit(bola komponentdan turib ota komponentga ta'sir qilish)dan foydalangan holda ota / bola o'zaro ta'sirlarini amalga oshirishga harakat qilishingiz kerak.
 
 <div class="composition-api">
 
-An exception here is that components using `<script setup>` are **private by default**: a parent component referencing a child component using `<script setup>` won't be able to access anything unless the child component chooses to expose a public interface using the `defineExpose` macro:
+`<script setup>` ni **tabiiy holatda shaxsiy(by default private)** qilgan holatda foydalanayotgan komponentlar uchun istisno bor: bola komponent `defineExpose` `compiler` idan foydalanib umumiy interfeysni ochishni tanlamagunicha, ota komponent `<script setup>` dan foydalanayotgan bola komponentga bog'lanayotib bola komponentning hech narsaniga kira olmaydi - foydalana olmaydi:
```

Comment:
Shu joyiga tushunmadim tarjimani
![image](https://user-images.githubusercontent.com/12369295/172147804-9e933ff4-c3e2-4e92-8da0-5d2c8e049155.png)

Reply:
hech narsasiga kira olmaydi dimoqchi bo'lganmisiz?

Reply:
ha shunday compilerni tarjima qilinadimi yoki o'zligicha qolaversinmi?

Reply:
To'g'rilandi.